app-frontend/src/app/components/mosaicParameters/mosaicScenesParameters.controller.js

```diff
@@ -0,0 +1,143 @@
+import _ from 'lodash';
+
+export default class MosaicScenesParametersController {
+    constructor( // eslint-disable-line max-params
+        $state, $scope, mapService
+    ) {
+        'ngInject';
+        this.$scope = $scope;
+        this.$state = $state;
+        this.getMap = () => mapService.getMap('project');
+    }
+
+    $onInit() {
+        this.mosaicType = 'Manual';
+        this.mosaicOptions = [
+            {name: 'Cloudiness Level'},
+            {name: 'Acquisition Date'},
+            {name: 'Upload Date'}
+        ];
+
+        this.extentBounds = {};
+        this.extentType = 'sceneExtent';
+    }
+
+    $onDestroy() {
+        this.getMap().then((map) => {
+            this.listeners.forEach((listener) => {
+                map.off(listener);
+            });
+        });
+        this.removeDrawControl();
+    }
+
+    /** Gets bounds of all scenes combined
+     *
+     * @returns {L.LatLngBounds} Bounds for combined scenes
+     */
+    getSceneBounds() {
+        this.getMap().then((map) => {
+            let layerBounds = _.map(map.getLayers('scenes'), function (layer) {
+                return layer.options.bounds;
+            });
+            return _.reduce(layerBounds, function (originalBnds, newBnds) {
+                return originalBnds.extend(newBnds);
+            });
+        });
+    }
+
+    /** Adds draw control to map
+     *
+     * @returns {undefined}
+     */
+    addDrawControl() {
+        this.getMap().then((map) => {
+            let extentLayer = L.geoJSON(null, {
+                style: () => {
+                    return {
+                        dashArray: '10, 10',
+                        weight: 2,
+                        fillOpacity: 0.2
+                    };
+                }
+            });
+
+            map.addLayer('extent', extentLayer);
+
+            let drawCtlOptions = {
+                position: 'topleft',
+                draw: {
+                    polyline: false,
+                    marker: false,
+                    circle: false,
+                    polygon: false,
+                    rectangle: {
+                        shapeOptions: {
+                            dashArray: '10, 10',
+                            weight: 2,
+                            fillOpacity: 0.2
+                        }
+                    }
+                },
+                edit: {
+                    featureGroup: extentLayer,
+                    remove: true
+                }
+            };
+            this.drawControl = new L.Control.Draw(drawCtlOptions);
+            map.map.addControl(this.drawControl);
+            this.listeners = [
```

Comment:
we should add an edit listener, even if it's just an empty method with a // TODO in it.